src/lightning/app/source_code/local.py

```diff
@@ -28,9 +28,13 @@
 class LocalSourceCodeDir:
     """Represents the source code directory and provide the utilities to manage it."""
 
-    cache_location: Path = Path.home() / ".lightning" / "cache" / "repositories"
-
     def __init__(self, path: Path, ignore_functions: Optional[List[_IGNORE_FUNCTION]] = None) -> None:
+        if "LIGHTNING_VSCODE_WORKSPACE" in os.environ:
+            # Don't use home to store the tar ball. This won't play nice with symlinks
+            self.cache_location: Path = Path(os.path.join("/tmp", ".lightning", "cache", "repositories"))
```

Comment:
```suggestion
            self.cache_location: Path = Path("/tmp", ".lightning", "cache", "repositories")
```
the join is not necessary